game_engine.hpp

```diff
@@ -14,11 +15,16 @@ template <std::size_t RowCount, std::size_t ColumnCount>
 class game_engine
 {
 public:
-    CONSTEXPR game_engine(const board<RowCount, ColumnCount>& board, int score, int moves, long long epoch_ms) : 
-        board_(board),
+    CONSTEXPR game_engine(const std::tuple< std::array<std::array<candy, ColumnCount>, RowCount>, int, int, bool, int, int > board_extended, int score, int moves, long long epoch_ms) :
+        board_(std::get<0>(board_extended)),
         score_(score),
         moves_(moves),
-        rg_(static_cast<std::uint16_t>(epoch_ms))
+        rg_(static_cast<std::uint16_t>(epoch_ms)),
+        hovered_x_(std::get<1>(board_extended)),
+        hovered_y_(std::get<2>(board_extended)),
+        any_selected_(std::get<3>(board_extended)),
```

Comment:
GCC with -Wall might complain that the order of the initializer list is not same as the members.

Reply:
Turns out it complains about more than just order of initializers! :-) I'll dedicate a branch for warnings clean up.

Reply:
Right!